zanata-war/src/main/java/org/zanata/dao/ProjectDAO.java

```diff
@@ -405,37 +409,36 @@ public HPerson getLastTranslator(HProject project) {
-                                "from HProject p " +
-                                "where :maintainer in elements(p.maintainers) " +
-                                "and p.status <> " +
-                                "'" + EntityStatus.OBSOLETE.getInitial() + "' " +
-                                "and (p.name like :filter " +
-                                    "or p.slug like :filter) " +
-                                "order by p.name")
-                        .setParameter("maintainer", maintainer)
-                        .setParameter("filter", "%" + sqlFilter + "%")
-                        .setFirstResult(firstResult)
-                        .setMaxResults(maxResults);
+        Query q = getSession().createQuery(
+                "select m.project from HProjectMember m " +
+                        "where m.person = :maintainer " +
+                        "and m.role = :role " +
+                        "and m.project.status <> :obsolete " +
+                        "and (m.project.name like :filter " +
+                        "or m.project.slug like :filter) " +
+                        "order by m.project.name")
+                .setParameter("maintainer", maintainer)
+                .setParameter("role", ProjectRole.Maintainer)
+                .setParameter("obsolete", EntityStatus.OBSOLETE)
+                .setParameter("filter", "%" + sqlFilter + "%")
+                .setFirstResult(firstResult)
+                .setMaxResults(maxResults);
         return q.list();
     }
 
     public int getMaintainedProjectCount(HPerson maintainer, String filter) {
         final String sqlFilter = filter == null ? "" : filter;
-        Query q =
-                getSession()
-                        .createQuery(
-                                "select count(p) from HProject p " +
-                                "where :maintainer in elements(p.maintainers) " +
-                                "and p.status <> " +
-                                "'" + EntityStatus.OBSOLETE.getInitial() + "' " +
-                                "and (p.name like :filter " +
-                                    "or p.slug like :filter) "
-                        )
-                        .setParameter("maintainer", maintainer)
-                        .setParameter("filter", "%" + sqlFilter + "%");
+        Query q = getSession().createQuery(
+                "select count(m) from HProjectMember m " +
+                        "where m.person = :maintainer " +
+                        "and m.role = :role " +
+                        "and m.project.status <> :obsolete " +
+                        "and (m.project.name like :filter " +
+                        "or m.project.slug like :filter)")
+                .setParameter("maintainer", maintainer)
+                .setParameter("role", ProjectRole.Maintainer)
+                .setParameter("obsolete", EntityStatus.OBSOLETE)
+                .setParameter("filter", "%" + sqlFilter + "%");
```

Comment:
Fair enough. I just thought the project search at the top banner is case
insensitive but this is not. We could change this later

On Thu, Aug 27, 2015 at 4:22 PM, David Mason <notifications@github.com>
wrote:

> In zanata-war/src/main/java/org/zanata/dao/ProjectDAO.java
> <https://github.com/zanata/zanata-server/pull/948#discussion_r38065418>:
>
> > -                                "and (p.name like :filter " +
> > -                                    "or p.slug like :filter) "
> > -                        )
> > -                        .setParameter("maintainer", maintainer)
> > -                        .setParameter("filter", "%" + sqlFilter + "%");
> > +        Query q = getSession().createQuery(
> > +                "select count(m) from HProjectMember m " +
> > +                        "where m.person = :maintainer " +
> > +                        "and m.role = :role " +
> > +                        "and m.project.status <> :obsolete " +
> > +                        "and (m.project.name like :filter " +
> > +                        "or m.project.slug like :filter)")
> > +                .setParameter("maintainer", maintainer)
> > +                .setParameter("role", ProjectRole.Maintainer)
> > +                .setParameter("obsolete", EntityStatus.OBSOLETE)
> > +                .setParameter("filter", "%" + sqlFilter + "%");
>
> Just the filters that were already there - the only real change here is to
> make the query work with the new database structure for maintainers, so I
> tried to keep the behaviour the same.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/zanata/zanata-server/pull/948/files#r38065418>.
>



-- 
Patrick Huang
Senior Software Engineer
Engineering - Internationalisation
Red Hat, Asia-Pacific Pty Ltd
Level 1, 193 North Quay
Brisbane 4000
Office: +61 7 3514 8278
Fax: +61 7 3514 8199
IRC: pahuang
github: github.com/huangp
Website: www.redhat.com